senders/client_factory.go

```diff
@@ -67,15 +67,26 @@ func CreateConfig(wfURL string, setters ...Option) (*configuration, error) {
 		return nil, err
 	}
 
-	if !strings.HasPrefix(strings.ToLower(u.Scheme), "http") {
-		return nil, fmt.Errorf("invalid scheme '%s' in '%s', only 'http' is supported", u.Scheme, u)
-	}
-
 	if len(u.User.String()) > 0 {
 		cfg.Token = u.User.String()
 		u.User = nil
 	}
 
+	switch strings.ToLower(u.Scheme) {
+	case "http":
+		if cfg.Token != "" {
+			cfg.MetricsPort = 80
+			cfg.TracesPort = 80
```

Comment:
There are four places in this method we set `MetricsPort` and `TracesPort` (some of the conditionally). Can we DRY up setting the defaults via helper function, where it determines if it's DI or Proxy and HTTP or HTTPS?

`func defaultTelemetryPorts(...) (metricsPort, tracesPort int)`

Reply:
I added a method to configuration to set the default port.